### PR TITLE
Fly model

### PR DESCRIPTION
**Pull request by andrewkern**
okay guys here is a module for _Drosophila melanogaster_. With the help of @jradrion I've packaged up the genetic maps estimated from Comeron et al (2012) to play nice with the `stdpopsim` conventions. I've implemented the demographic model from Sheehan and Song (2016) which is a three epoch model estimated from African _Drosophila melanogaster_ data. I have also implemented the required unit tests following the conventions that @jeromekelleher established for the _Homo sapiens_ module. 

One issue is that the genetic maps are currently sitting on a UO server. They could easily be moved someplace else at a later date, I'm just not sure where might be better. Something to consider anyway.

### CODE REVIEW

File: tests/test_drosophila_melanogaster.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.assertEqual(len(genome.chromosomes), 8)
+
+    def test_str(self):
+        s = str(drosophila_melanogaster.genome)
+        self.assertGreater(len(s), 0)
+        self.assertIsInstance(s, str)
+
+    def test_chromosome_lengths(self):
+        genome = drosophila_melanogaster.genome
+        # Numbers from https://www.ncbi.nlm.nih.gov/grc/human/data
```
Comment (jeromekelleher) [inline tests/test_drosophila_melanogaster.py]: This comment is misleading, right? Should just be "Numbers from dm6"

File: stdpopsim/drosophila_melanogaster.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        t_2_coal = 5.0
+        # estimates from the ANN
+        N_R = 544200
+        N_B = 145300
+        N_A = 652700
+        # Times are provided in 4N_ref generations, so we convert into generations.
+        # generation_time = 10 / year
+        t_1 = t_1_coal * 4 * N_ref
+        t_2 = t_2_coal * 4 * N_ref
+        # Population IDs correspond to their indexes in the population
```
Comment (jeromekelleher) [inline stdpopsim/drosophila_melanogaster.py]: This comment is misleading. I would delete it.

File: stdpopsim/drosophila_melanogaster.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+for line in _chromosome_data.splitlines():
+    name, length = line.split()[:2]
+    _chromosomes.append(genomes.Chromosome(
+        name=name, length=int(length),
+        mean_mutation_rate=8.4e-9,  # WRONG!, underestimate used in S&S
+        mean_recombination_rate=8.4e-9))  # WRONG, underestimate used in S&S!
+
+
+#: :class:`stdpopsim.Genome` definition for D. melanogaster. Chromosome length data is
+#: based on `dm6 <https://www.ncbi.nlm.nih.gov/assembly/GCF_000001215.4/>`_.
```
Comment (jeromekelleher) [inline stdpopsim/drosophila_melanogaster.py]: I tweaked the documentation a bit here to have a link (this gets rendered on RTD). Is this link correct?

File: tests/test_drosophila_melanogaster.py (hunk 1 above)
Comment (andrewkern) [reply tests/test_drosophila_melanogaster.py]: yep. ugh didn't clean out the comments after I started working from the Homo_sapiens example

File: stdpopsim/drosophila_melanogaster.py (hunk 3 above)
Comment (andrewkern) [reply stdpopsim/drosophila_melanogaster.py]: yep this is the correct link

File: stdpopsim/drosophila_melanogaster.py (hunk 2 above)
Comment (andrewkern) [reply stdpopsim/drosophila_melanogaster.py]: fixed

**Comment by andrewkern** [comment]
Okay having made those clean ups and with approval from @jeromekelleher, I'm gonna merge this PR

**Pull request merged**